src/EnergyPlus/HeatBalFiniteDiffManager.cc

```diff
@@ -236,6 +236,7 @@ namespace HeatBalFiniteDiffManager {
                 // Load the material derived type from the input data.
                 MaterNum = UtilityRoutines::FindItemInPtrList(MaterialNames(1), state.dataMaterial->Material);
                 auto const *thisMaterial = state.dataMaterial->Material(MaterNum);
+                auto const *thisMaterialChild = dynamic_cast<const Material::MaterialChild *>(thisMaterial);
```

Comment:
`dynamic_cast` will crash I believe if `thisMaterial` is `nullptr`. This shortcut has to come after the `if (MaterNum == 0)` test.

Reply:
indeed. just moved it to after the MaterialNum == 0 check